### PR TITLE
Add formula for Jetbrains Fleet

### PR DESCRIPTION
**Pull request by Kwadz**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [ ] `brew audit --cask --online <cask>` is error-free.
- [ ] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --new-cask <cask>` worked successfully.
- [ ] `brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/fleet.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+cask "fleet" do
+  arch arm: "-aarch64"
+
+  version "2022.2.3,222.4345.15"
+  sha256 arm:   "0dee8fe654cccdafa73b65da1a2ef844401a9438ecee726fe6f6af1f09d07c38",
+         intel: "8dbe5cd8e31c7f6bc6795db6946e2430c82f0aa2c13e7805c40733428b02241d"
+
+  url "https://download.jetbrains.com/webide/PhpStorm-#{version.csv.first}#{arch}.dmg"
```
Comment (SMillerDev) [inline Casks/fleet.rb:8]: This is missing a fleet specific URL

File: Casks/fleet.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+cask "fleet" do
+  arch arm: "-aarch64"
+
+  version "2022.2.3,222.4345.15"
```
Comment (SMillerDev) [inline Casks/fleet.rb:4]: This is not the version of Fleet

File: Casks/fleet.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      end
+    end
+  end
+
+  zap trash: [
+    "~/Library/Application Support/PhpStorm#{version.major_minor}",
+    "~/Library/Caches/PhpStorm#{version.major_minor}",
+    "~/Library/Logs/PhpStorm#{version.major_minor}",
+    "~/Library/Preferences/jetbrains.phpstorm.*.plist",
+    "~/Library/Preferences/PhpStorm#{version.major_minor}",
```
Comment (SMillerDev) [inline Casks/fleet.rb:41]: Needs to be updated for Fleet

**Comment by Kwadz** [comment]
It seems the .dmg file is not yet available, according to the [download page](https://www.jetbrains.com/fleet/download/#section=mac):

> Fleet is free to use during the public preview and currently available through the [Toolbox App](https://www.jetbrains.com/toolbox-app/) only.

I didn't see. I guess I have to wait before completing this PR, sorry...

**Comment by p-linnane** [comment]
@Kwadz Once it becomes available as a standalone app we can revisit this.

**Pull request closed without merging**